samples/snippets/schema.py

```diff
@@ -189,24 +191,119 @@ def create_topic_with_schema(project_id, topic_id, schema_id, encoding):
     # [END pubsub_create_topic_with_schema]
 
 
-def publish_avro_records(project_id, topic_id):
+def publish_avro_records(project_id, topic_id, avsc_file):
     """Pulbish a BINARY or JSON encoded message to a topic configured with an Avro schema."""
     # [START pubsub_publish_avro_records]
-    pass
+    from avro.io import BinaryEncoder, DatumWriter
+    import avro
+    import io
+    import json
+    from google.api_core.exceptions import NotFound, InvalidArgument
+    from google.cloud.pubsub import PublisherClient
+    from google.pubsub_v1.types import Encoding
+
+    # TODO(developer): Replace these variables before running the sample.
+    # project_id = "your-project-id"
+    # topic_id = "your-topic-id"
+    # avsc_file = "path/to/an/avro/schema/file/(.avsc)/formatted/in/json"
+
+    publisher_client = PublisherClient()
+    topic_path = publisher_client.topic_path(project_id, topic_id)
+
+    # Prepare to write Avro records to the binary output stream.
+    avro_schema = avro.schema.parse(open(avsc_file, "rb").read())
+    writer = DatumWriter(avro_schema)
+    bout = io.BytesIO()
+
+    # Prepare some data using a Python dictionary that matches the Avro schema
+    record = {"name": "Alaska", "post_abbr": "AK"}
+
+    try:
+        # Get the topic encoding type.
+        topic = publisher_client.get_topic(request={"topic": topic_path})
+        encoding = topic.schema_settings.encoding
+
+        # Encode the data according to the message serialization type.
+        if encoding == Encoding.BINARY:
+            encoder = BinaryEncoder(bout)
+            writer.write(record, encoder)
+            data = bout.getvalue()
+
+            print("Preparing a binary-encoded message.")
+        elif encoding == Encoding.JSON:
+            data = json.dumps(record).encode("utf-8")
+            print("Preparing a JSON-encoded message.")
+        else:
+            raise InvalidArgument(ValueError)
```

Comment:
(applies to multiple places)

The first argument to `InvalidArgument` should be a string (error messages). Also, `InvalidArgument` inherits from `GoogleAPICallError`, which does not seem like a correct exception type, as it's not a result of an API call.

BTW, the only way to hit this branch is if topic's schema settings encoding is `ENCODING_UNSPECIFIED`, but that's the info retrieved from the server? (i.e. not something we would pass as an argument)

Reply:
I rewrote this part to exit(0). Is that better?

Reply:
The approach seems reasonable IMO. :+1: